Keithley_2700Sup/parse_channel_timestamps.c

```diff
@@ -0,0 +1,118 @@
+#include <registryFunction.h>
+#include <epicsExport.h>
+#include "aSubRecord.h"
+#include "stdlib.h"
+#include <stdio.h>
+
+/*
+*	Keithley 2700 Timestamp Data Parser
+*
+* 	Takes data from waveform record BUFFER:READINGS, finds the TIMESTAMP value and 
+*	puts it into the correct Channel. 
+*
+*	VALA (value A) = channel 101 reading = CHNL:101:TIMESTAMP
+*
+*	"Why are there two C scripts that do very similar things (parse_channel_timestamps.c 
+*	and parse_channel_readings.c)?"
+*
+*	Good question. aSub records have a limited number of outputs and so all of the channel
+* 	:READING and :TIMESTAMP PVs couldn't be outputs for one record. An alternate solution could 
+*	be to write the channel values to a waveform PV for each channel (e.g. CHNL:101:RAWVALS) and 
+*	have a calc record in each channel which splits the waveform into :READING and :TIMESTAMP PVs.
+*
+*/
+
+long i;
+double *a;
+double *vala, *valb,*valc,*vald,*vale,*valf,*valg,*valh,*vali,*valj,*valk,
+*vall,*valm,*valn,*valo,*valp,*valq,*valr,*vals,*valt;
+
+static long parse_channel_timestamps(aSubRecord *prec) {
```

Comment:
This is functionally identical as far as I can see to `parse_channel_readings` except the set value is different. Find a way to eliminate the code duplication.